provider/provider.go

```diff
@@ -86,12 +101,22 @@ func NewTile(z, x, y, buf, srid uint) Tile {
 
 // Extent returns the extent of the tile
 func (tile *tile_t) Extent() (ext *geom.Extent, srid uint64) {
-	return tile.Extent3857(), 3857
+	if ext, ok := slippy.Extent(webmercatorGrid, &tile.Tile); ok {
+		return ext, 3857
+	}
+
+	log.Error("Could not generate valid extent for tile.", tile)
+	return &geom.Extent{}, 3857
 }
 
 // BufferedExtent returns a the extent of the tile, with the define buffer
 func (tile *tile_t) BufferedExtent() (ext *geom.Extent, srid uint64) {
-	return tile.Extent3857().ExpandBy(slippy.Pixels2Webs(tile.Z, tile.buffer)), 3857
+	ext, _ = tile.Extent()
+	// WARNING: slippy.PixelsToNative() is misnamed. It doesn't convert the given number of pixels into
+	// anything--the pixels arguments isn't even used (hence set to 0 here). Instead, slippy.PixelsToNative()
+	// return the ratio of pixels to projected units at the given zoom. We multiply its return value by
+	// the pixel count in tile.buffer to get the expected conversion.
```

Comment:
Let's create an issue around this to get the function named correctly.

Reply:
I've opened https://github.com/go-spatial/geom/issues/128 and https://github.com/go-spatial/geom/issues/129.